Compatibility with Elasticsearch 8.11

### Preconditions

- Magento version: 2.4.6-p3
- ElasticSuite version: 2.11.4.3
- Environment: Developer mode

On Debian 11, Elasticsearch 8.11.4

### Steps to reproduce

Good evening,

I did an update from Elasticsearch 7.17 to the last (8.11.X series).
Before doing this update, I followed on Kibana the "Upgrade Assistant" to prepare the upgrade.

I have an issue, as I know, all products lists are empty.

I checked file log from ElasticSearch, this is what I have :

> [2024-01-11T16:46:25,189][WARN ][r.suppressed             ] [amadeco-node1] path: /magento2_default_catalog_product/_search, params: {index=magento2_default_catalog_product}
> org.elasticsearch.action.search.SearchPhaseExecutionException: all shards failed
> [...]
> Caused by: org.elasticsearch.search.aggregations.AggregationExecutionException: Invalid aggregation order path [_term]. The provided aggregation [_term] either does not exist, or is a pipeline aggregation and cannot be used to>
> [...]

As you can see, it seems come from the order field _term in 
[elasticsuite/src/module-elasticsuite-core/Search/Request/BucketInterface.php](https://github.com/Smile-SA/elasticsuite/blob/f706e61a7bcb90db3a36d87b51585d13a865fb16/src/module-elasticsuite-core/Search/Request/BucketInterface.php#L38)
`    const SORT_ORDER_TERM      = '_term';`

Indeed, when I replace this field __term_ by another from Elasticsearch (cf. [https://www.elastic.co/guide/en/elasticsearch/reference/current/sort-search-results.html#_sort_values_2](https://www.elastic.co/guide/en/elasticsearch/reference/current/sort-search-results.html#_sort_values_2)), queries are working right.

I did a long search in elasticsearch docs (7.11.X and 8.11.X) but I have no reference at all if this sort value has been deprecated.
I even tried to reinstall totaly Elasticsearch (config files, data files, etc.), thinking this issue comes from my installation, but from a fresh install, I have still this problem.

As I have no experience in ElasticSeach, maybe you know something I don't ?

I saw on some issues page, you seem to have tested your extension on ElasticSuite 8.

Thank you

--- a/src/module-elasticsuite-catalog/Model/Attribute/Source/FilterSortOrder.php
+++ b/src/module-elasticsuite-catalog/Model/Attribute/Source/FilterSortOrder.php
@@ -35,7 +35,7 @@ class FilterSortOrder implements OptionSourceInterface
         return [
             ['value' => BucketInterface::SORT_ORDER_COUNT, 'label' => __('Result count')],
             ['value' => BucketInterface::SORT_ORDER_MANUAL, 'label' => __('Admin sort')],
-            ['value' => BucketInterface::SORT_ORDER_TERM, 'label' => __('Name')],
+            ['value' => BucketInterface::SORT_ORDER_TERM_DEPRECATED, 'label' => __('Name')],
             ['value' => BucketInterface::SORT_ORDER_RELEVANCE, 'label' => __('Relevance')],
         ];
     }

--- a/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Request/Aggregation/Builder/Term.php
+++ b/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Request/Aggregation/Builder/Term.php
@@ -48,7 +48,7 @@ class Term implements BuilderInterface
             $aggregation['terms']['order'] = $bucket->getSortOrder();
         } elseif (in_array($bucket->getSortOrder(), [$bucket::SORT_ORDER_COUNT, $bucket::SORT_ORDER_MANUAL])) {
             $aggregation['terms']['order'] = [$bucket::SORT_ORDER_COUNT => SortOrderInterface::SORT_DESC];
-        } elseif ($bucket->getSortOrder() == $bucket::SORT_ORDER_TERM) {
+        } elseif (in_array($bucket->getSortOrder(), [$bucket::SORT_ORDER_TERM, $bucket::SORT_ORDER_TERM_DEPRECATED])) {
             $aggregation['terms']['order'] = [$bucket::SORT_ORDER_TERM => SortOrderInterface::SORT_ASC];
         } elseif ($bucket->getSortOrder() == $bucket::SORT_ORDER_RELEVANCE && !$bucket->isNested()) {
             $aggregation['aggregations']['termRelevance'] = ['avg' => ['script' => $bucket::SORT_ORDER_RELEVANCE]];

--- a/src/module-elasticsuite-core/Search/Request/BucketInterface.php
+++ b/src/module-elasticsuite-core/Search/Request/BucketInterface.php
@@ -35,9 +35,10 @@ interface BucketInterface extends \Magento\Framework\Search\Request\BucketInterf
     const TYPE_METRIC           = 'metricBucket';
 
     const SORT_ORDER_COUNT     = '_count';
-    const SORT_ORDER_TERM      = '_term';
+    const SORT_ORDER_TERM      = '_key';
     const SORT_ORDER_RELEVANCE = "_score";
     const SORT_ORDER_MANUAL    = "_manual";
+    const SORT_ORDER_TERM_DEPRECATED = '_term';
 
     /**
      * @var integer

--- a/src/module-elasticsuite-core/Test/Unit/Search/Adapter/Elasticsuite/Request/Aggregation/Builder/TermTest.php
+++ b/src/module-elasticsuite-core/Test/Unit/Search/Adapter/Elasticsuite/Request/Aggregation/Builder/TermTest.php
@@ -53,7 +53,46 @@ class TermTest extends \PHPUnit\Framework\TestCase
     public function testAplhabeticSortOrderTermAggregationBuild()
     {
         $aggBuilder = $this->getAggregationBuilder();
-        $termBucket = new TermBucket('aggregationName', 'fieldName', [], [], [], null, null, null, 1, TermBucket::SORT_ORDER_TERM);
+        $termBucket = new TermBucket(
+            'aggregationName',
+            'fieldName',
+            [],
+            [],
+            [],
+            null,
+            null,
+            null,
+            1,
+            TermBucket::SORT_ORDER_TERM
+        );
+
+        $aggregation = $aggBuilder->buildBucket($termBucket);
+
+        $this->assertArrayHasKey('terms', $aggregation);
+        $this->assertEquals('fieldName', $aggregation['terms']['field']);
+        $this->assertEquals([TermBucket::SORT_ORDER_TERM => SortOrderInterface::SORT_ASC], $aggregation['terms']['order']);
+    }
+
+    /**
+     * Test the standard term aggregation building sorted by alphabetic order using the deprecated sort order.
+     *
+     * @return void
+     */
+    public function testAplhabeticSortOrderTermDeprecatedAggregationBuild()
+    {
+        $aggBuilder = $this->getAggregationBuilder();
+        $termBucket = new TermBucket(
+            'aggregationName',
+            'fieldName',
+            [],
+            [],
+            [],
+            null,
+            null,
+            null,
+            1,
+            TermBucket::SORT_ORDER_TERM_DEPRECATED
+        );
 
         $aggregation = $aggBuilder->buildBucket($termBucket);
 

